pkg/datapath/linux/node_ids.go

```diff
@@ -274,6 +284,13 @@ func (n *linuxNodeHandler) RestoreNodeIDs() {
 	}
 
 	n.registerNodeIDAllocations(nodeIDs)
+	for ip := range incorrectNodeIDs {
+		if err := n.unmapNodeID(ip); err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.IPAddr: ip,
+			}).Warn("Failed to remove a incorrect node IP to node ID mapping")
+		}
+	}
 	log.Infof("Restored %d node IDs from the BPF map", len(nodeIDs))
 }
 
```

Comment:
(*Commenting in random place.*)

Could we describe the user impact in the release note? AFAICS, without Mutual Auth and IPsec, the only impact are superfluous error messages in logs. With IPsec, the failure to allocate node IDs would impact connectivity to new nodes in the cluster(mesh). With Mutual Auth, I'm unsure of the impact. Probably easier to describe each situation on a new line.